src/modules/75-cf/components/CreateFlagWizard/__tests__/FlagWizard.test.tsx

```diff
@@ -6,47 +6,100 @@
+        {...props}
+      />
+    </TestWrapper>
+  )
+}
+
 describe('FlagWizard', () => {
+  const createFlagMutateMock = jest.fn()
+  const useCreateFeatureFlagMock = jest.spyOn(cfServiceMock, 'useCreateFeatureFlag')
+
+  beforeEach(() => {
+    jest
+      .spyOn(cfServiceMock, 'useGetAllTags')
+      .mockReturnValue({ loading: false, data: [], refetch: jest.fn(), error: null } as any)
+
+    useCreateFeatureFlagMock.mockReturnValue({
+      mutate: createFlagMutateMock,
+      loading: false,
+      error: null
+    } as any)
+  })
+
   test('it should fire telementary event when completed created flag', async () => {
     jest.spyOn(cfServiceMock, 'useGetGitRepo').mockReturnValue({ loading: false, data: { repoSet: true } } as any)
 
-    render(
-      <TestWrapper
-        path="/account/:accountId/cf/dashboard/orgs/:orgIdentifier/projects/:projectIdentifier"
-        pathParams={{ accountId: 'dummy', orgIdentifier: 'dummy', projectIdentifier: 'dummy' }}
-      >
-        <FlagWizard
-          flagTypeView={FlagTypeVariations.booleanFlag}
-          environmentIdentifier="nonProduction"
-          toggleFlagType={jest.fn()}
-          hideModal={jest.fn()}
-          goBackToTypeSelections={jest.fn()}
-        />
-      </TestWrapper>
-    )
-
-    await userEvent.type(screen.getByPlaceholderText('cf.creationModal.aboutFlag.ffNamePlaceholder'), 'TEST_FLAG')
-
-    fireEvent.click(screen.getByText('next'))
+    renderComponent()
+
+    await waitFor(() => {
+      userEvent.type(screen.getByPlaceholderText('cf.creationModal.aboutFlag.ffNamePlaceholder'), 'TEST_FLAG')
+      userEvent.click(screen.getByText('next'))
+    })
 
     expect(trackEventMock).toHaveBeenCalled()
   })
+
+  test('it should create a flag with the correct payload', async () => {
+    const isTaggingFlagOn = true
+    const newTag = 'my new tag'
+    const newFlag = 'my new flag'
+
+    renderComponent(isTaggingFlagOn)
+
+    const flagsInput = screen.getByPlaceholderText('cf.creationModal.aboutFlag.ffNamePlaceholder')
+    const tagsDropdownInput = screen.getByPlaceholderText('- tagsLabel -')
+
+    await userEvent.type(flagsInput, newFlag)
+    expect(flagsInput).toHaveValue(newFlag)
+
+    await userEvent.type(tagsDropdownInput, newTag)
+    expect(tagsDropdownInput).toHaveValue(newTag)
+
+    await userEvent.click(screen.getByRole('button', { name: 'next' }))
+
+    await userEvent.click(screen.getByRole('button', { name: 'cf.creationModal.saveAndClose' }))
+    expect(useCreateFeatureFlagMock).toHaveBeenCalledWith({
+      queryParams: {
+        accountIdentifier: 'dummy',
+        environmentIdentifier: 'nonProduction',
+        orgIdentifier: 'dummy'
+      }
+    })
+
+    // redirects to flag details page of the newly created flag
+    await waitFor(() => {
+      expect(screen.getByTestId('location')).toHaveTextContent(
+        '/account/dummy/cf/orgs/dummy/projects/dummy/feature-flags/my_new_flag?activeEnvironment=nonProduction'
+      )
+    })
```

Comment:
Nothing wrong with using `waitFor`, but `findBy` is great for things like this.
```suggestion
    expect(await screen.findByTestId('location')).toHaveTextContent(
      '/account/dummy/cf/orgs/dummy/projects/dummy/feature-flags/my_new_flag?activeEnvironment=nonProduction'
    )
```